### PR TITLE
Fix: Home screen - The json data is incomplete. [ED-14140]

### PR DESCRIPTION
**Pull request by hein-obox**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: modules/home/transformations/create-new-page-url.php [hunk 1]
```diff
@@ -10,9 +10,9 @@
 
 class Create_New_Page_Url extends Transformations_Abstract {
 
-	public function transform(): array {
-		$this->home_screen_data['create_new_page_url'] = Plugin::$instance->documents->get_create_new_post_url( 'page' );
+	public function transform( $home_screen_data ): array {
```
Comment (mykytamurzin) [inline modules/home/transformations/create-new-page-url.php:13]: `array $home_screen_data`

File: modules/home/transformations/filter-plugins.php [hunk 2]
```diff
@@ -9,10 +9,8 @@
 
 class Filter_Plugins extends Transformations_Abstract {
 
-	public function transform(): array {
-		$this->set_add_on_installation_status();
-
-		return $this->home_screen_data;
+	public function transform( $home_screen_data ): array {
+		return $this->set_add_on_installation_status( $home_screen_data );
```
Comment (mykytamurzin) [inline modules/home/transformations/filter-plugins.php:13]: what if ? 
```
$home_screen_data['add_ons']['repeater'] = $this->get_add_ons_status($home_screen_data['add_ons']['repeater']);

return $home_screen_data;
```

File: modules/home/transformations/create-new-page-url.php (hunk 1 above)
Comment (hein-obox) [reply modules/home/transformations/create-new-page-url.php:13]: Updated.

File: modules/home/transformations/filter-plugins.php [hunk 3]
```diff
@@ -9,10 +9,8 @@
 
 class Filter_Plugins extends Transformations_Abstract {
 
-	public function transform(): array {
-		$this->set_add_on_installation_status();
-
-		return $this->home_screen_data;
+	public function transform( $home_screen_data ): array {
```
[the remark this answers is not in the document]
Comment (hein-obox) [reply modules/home/transformations/filter-plugins.php:12]: Updated.

File: modules/home/transformations/filter-plugins.php (hunk 2 above)
Comment (hein-obox) [reply modules/home/transformations/filter-plugins.php]: Updated.

**Review by murzin-nikita — DISMISSED** [review]

File: modules/home/transformations/filter-plugins.php [hunk 4]
```diff
@@ -21,19 +21,20 @@ private function get_installed_plugins(): array {
...
 	}
 
-	private function set_add_on_installation_status() {
-		$add_ons = $this->home_screen_data['add_ons']['repeater'];
+	private function get_add_ons_installation_status( array $add_ons ): array {
 		$index = 0;
 
 		foreach ( $add_ons as $add_on ) {
 			$is_plugin = array_key_exists( 'file_path', $add_on );
 			$is_installed_plugin = $is_plugin && in_array( $add_on['file_path'], $this->get_installed_plugins() );
```
Comment (TzviRabinovitch) [inline modules/home/transformations/filter-plugins.php:29]: maybe move this to inside the if below and don't check for is_plugin twice ?

Comment (TzviRabinovitch) [inline modules/home/transformations/filter-plugins.php:29]: do we need to run get_installed_plugins() for every iteration ?

Comment (hein-obox) [reply modules/home/transformations/filter-plugins.php:29]: @TzviRabinovitch The json is stored in the transient, and is only regenerated once an hour. Does that answer your question?

Comment (hein-obox) [reply modules/home/transformations/filter-plugins.php]: Updated.

**Review by TzviRabinovitch — DISMISSED** [review]

**Review by TzviRabinovitch — APPROVED** [review]

**Pull request merged**